src/lib/components/FilterPagePanel.svelte

```diff
@@ -80,15 +72,12 @@
     </div>
     <div class="flex justify-center">
         <div class="grid w-fit grid-cols-1 items-center gap-8 md:grid-cols-2 lg:grid-cols-3">
-            {#each filteredEventsInPage as { type, state, url, title, src, dates, description }}
-                <EventCard {src} alt="image of {title}" href={url}>
-                    <svelte:fragment>
-                        <p class="m-0">{state} - {type}</p>
-                        <h2 class="m-0">{title}</h2>
-                        <p class="m-0">{dates}</p>
-                        <p class="m-0 grow overflow-hidden">{description}</p>
-                    </svelte:fragment>
-                </EventCard>
+            {#each filteredCardsInPage as cardProps}
+                {#if 'state' in cardProps && 'dates' in cardProps}
+                    <svelte:component this={cardComponent} {...cardProps} />
```

Comment:
```suggestion
                {#if 'state' in cardProps}
                    <svelte:component this={EventCard} {...cardProps} />
```
We don't need both `'state' in cardProps` and `'dates' in cardProps` to satisfy the type checker. Just one of these would do.

Also, I suppose rather than `state` and `dates`, it might be better to change `title`—in both the `Event` and `Project` interfaces—to `event` and `project`, then base it off that.

Besides that, I'm still stumped over other possible ways we could've implemented this without the type-checker complaining. I think that if we wanted to do this properly, then we would have to redo this component entirely. To be honest, I don't think this is worth dealing with at the moment.